front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java

```diff
@@ -61,9 +61,17 @@
   private String applicationVersion;
 
   @Bean
-  public GcsStorageService googleCloudStorageService() {
+  public GcsStorageService defaultGoogleCloudStorageService() {
+    return googleCloudStorageService(null /*dataFilename*/);
+  }
+
+  private GcsStorageService googleCloudStorageService(String dataFilename) {
     GcsStorageService service;
-    service = new GcsStorageService(bucket, rootFolder, project, jsonPath, applicationVersion);
+    if (dataFilename == null) {
```

Comment:
Why not `!dataFilename` here? Seems like you would want the first case if it's an empty string as well.